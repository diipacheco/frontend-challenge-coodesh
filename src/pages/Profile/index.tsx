import React, { useState, useCallback, useEffect } from 'react';

import { IUser } from '../../hooks/users';

import Modal from '../../components/Modal';
import Header from '../../components/Header';

import { Container } from './styles';

const Profile: React.FC = () => {
  const [modalIsOpen, setIsOpen] = useState(false);
  const [localStorageFilteredUser, setLocalStorageFilteredUser] = useState(
    {} as IUser,
  );

  const closeModal = useCallback(() => {
    setIsOpen(false);
  }, []);

  useEffect(() => {
    setIsOpen(true);
    const filteredUser = localStorage.getItem('filteredUser') as string;
    setLocalStorageFilteredUser(JSON.parse(filteredUser));
  }, []);

  return (
    <Container>
      <Header />
      <Modal
        modalIsOpen={modalIsOpen}
        closeModal={closeModal}
        userInfo={localStorageFilteredUser}
      />
    </Container>
  );
};

export default Profile;
