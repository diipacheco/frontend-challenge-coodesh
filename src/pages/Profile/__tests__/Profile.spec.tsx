import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';

import history from '../../../services/history';
import Profile from '../index';

describe('<Profile />', () => {
  it('should match snapshots', () => {
    const wrapper = render(
      <Router history={history}>
        <Profile />
      </Router>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
