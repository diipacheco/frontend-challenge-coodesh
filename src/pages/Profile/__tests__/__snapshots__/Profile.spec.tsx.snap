// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should match snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="ReactModal__Body--open"
  >
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div>
      <div
        class="sc-gKseQn"
      >
        <header
          class="sc-eCstlR ecpwZt"
        >
          <h1>
            Pharma Inc
          </h1>
          <div
            class="sc-jSgvzq inNPAN"
          >
            <svg
              color="#A2B0BD"
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(162, 176, 189);"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </header>
      </div>
    </div>
    <div
      class="ReactModalPortal"
    >
      <div
        class="ReactModal__Overlay"
        style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px;"
      >
        <div
          aria-modal="true"
          class="ReactModal__Content"
          role="dialog"
          style="position: absolute; top: 50%; left: 50%; border: 1px solid #ccc; background: rgb(255, 255, 255); overflow: auto; border-radius: 4px; outline: none; padding: 20px; margin-right: -50%; transform: translate(-50%, -50%); width: 600px; height: 530px;"
          tabindex="-1"
        >
          <section
            class="sc-bdfBQB gqMTeA"
          >
            <button
              class="sc-hKgJUU cTQnKd"
              type="button"
            >
              <svg
                fill="currentColor"
                height="30"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </button>
            <div
              class="sc-dlfnuX gNwICt"
            />
            <h1>
              undefined undefined
            </h1>
            <div
              class="sc-gsTEea fXhQjI"
            >
              <strong>
                Email:
              </strong>
              <p />
            </div>
            <div
              class="sc-gsTEea fXhQjI"
            >
              <strong>
                Gender:
              </strong>
              <p />
            </div>
            <div
              class="sc-gsTEea fXhQjI"
            >
              <strong>
                Birth:
              </strong>
              <p>
                07/10/2021
              </p>
            </div>
            <div
              class="sc-gsTEea fXhQjI"
            >
              <strong>
                Phone:
              </strong>
              <p />
            </div>
            <div
              class="sc-gsTEea fXhQjI"
            >
              <strong>
                NAT:
              </strong>
              <p />
            </div>
            <div
              class="sc-gsTEea fXhQjI"
            >
              <strong>
                Address:
              </strong>
              <p>
                undefined, undefined - undefined - undefined
              </p>
            </div>
            <div
              class="sc-gsTEea fXhQjI"
            >
              <strong>
                ID:
              </strong>
              <p>
                undefined - undefined
              </p>
            </div>
            <div
              class="sc-gsTEea fXhQjI"
            >
              <a
                href="/profile/undefined"
              >
                URL
              </a>
            </div>
          </section>
        </div>
      </div>
    </div>
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
  </body>,
  "container": <div>
    <div
      class="sc-gKseQn"
    >
      <header
        class="sc-eCstlR ecpwZt"
      >
        <h1>
          Pharma Inc
        </h1>
        <div
          class="sc-jSgvzq inNPAN"
        >
          <svg
            color="#A2B0BD"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(162, 176, 189);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
